src/runtime/virtcontainers/hypervisor.go

```diff
@@ -894,17 +880,55 @@ func GetHypervisorPid(h Hypervisor) int {
 	return pids[0]
 }
 
-func generateVMSocket(id string, vmStogarePath string) (interface{}, error) {
-	vhostFd, contextID, err := utils.FindContextID()
-	if err != nil {
-		return nil, err
-	}
+// Kind of guest protection
+type guestProtection uint8
+
+const (
+	noneProtection guestProtection = iota
+
+	//Intel Trust Domain Extensions
+	//https://software.intel.com/content/www/us/en/develop/articles/intel-trust-domain-extensions.html
+	// Exclude from lint checking for it won't be used on arm64 code
+	tdxProtection
+
+	// AMD Secure Encrypted Virtualization
+	// https://developer.amd.com/sev/
+	// Exclude from lint checking for it won't be used on arm64 code
+	sevProtection
+
+	// IBM POWER 9 Protected Execution Facility
+	// https://www.kernel.org/doc/html/latest/powerpc/ultravisor.html
+	// Exclude from lint checking for it won't be used on arm64 code
+	pefProtection
+
+	// IBM Secure Execution (IBM Z & LinuxONE)
+	// https://www.kernel.org/doc/html/latest/virt/kvm/s390-pv.html
+	// Exclude from lint checking for it won't be used on arm64 code
+	seProtection
```

Comment:
Nit: If possible (?) might be worth sorting alphabetically but I see this is simply a verbatim move from another file.